### ISSUE TITLE
Results when a searchPhrase is longer than all targetPhrases (but includes an exact match substring)

### ISSUE THREAD

**Issue by brandrsn** [farzher/fuzzysort#77]
Hey @farzher, kudos for the awesome package. Quick question, if you don't mind?

When a `targetPhrase` is shorter in length than a `searchPhrase`, is there a way to surface this result, even if the extraneous characters of the `searchPhrase` are not found in the `targetPhrase`? That's a mouthful, so here's the use case:

```javascript
const variations = ["rutgers.edu", "rutgers edu", "rutgersedu", "edu.rutgers", "edu rutgers", "edurutgers"];
const options = {threshold: -Infinity, allowTypo: true};

const attempt1 = go("mail.rutgers.edu", variations, options); // No results
const attempt2 = go("mail rutgers edu", variations, options); // No results
const attempt3 = go("mailrutgersedu", variations, options); // No results
const attempt4 = go("edu.rutgers.mail", variations, options); // No results
const attempt5 = go("edu rutgers mail", variations, options); // No results
const attempt6 = go("edurutgersmail", variations, options); // No results
```
Above, the `mail` characters of the `searchPhrase` invalidate every `targetPhrase`, even through there's an exact match substring within it.

I've tried reversing the strings (to bring the exact match substring closer to the beginning of the `targetPhrase`), toggling `allowTypo`, and adding my `searchString`s to an array, but I can't coax a response out of it. I'd love even a low/poor match score, as opposed to no results.

Any thoughts?

Thanks!

**Comment by brandrsn** [comment]
I guess it's a similar result to this (as seen on [your test page](https://rawgit.com/farzher/fuzzysort/master/test.html)): 

List of all Sonic games:
![List of all Sonic games](https://geckoengage.s3.amazonaws.com/dump/Screenshot%202020-11-11%20at%2013.40.38.png)

Specific (misspelled) Sonic game:
![Specific (misspelled) Sonic game](https://geckoengage.s3.amazonaws.com/dump/Screenshot%202020-11-11%20at%2013.41.01.png)

**Comment by farzher** [comment]
fuzzysort can't do that, just like sublime text.

if you want to try to make fuzzysort do that you can perform multiple searches and combine the results like this https://github.com/farzher/fuzzysort/issues/58#issuecomment-602157414

but i don't see an obvious way to do what you want well

**Comment by brandrsn** [comment]
Yeah, I've used #58 and it doesn't get me where I want to be (I'm using it to combine all 6 attempts from my example code — in the [example above](https://github.com/farzher/fuzzysort/issues/77#issue-740771972), I just broke them out individually for clarity). Seems like merging/averaging a whole bunch of zeros results in zero!

I'm playing around with [another library](https://github.com/aceakash/string-similarity) too. Perhaps there's something I can do to fallback to the alternative when I can't get the desired results from fuzzysort.

Thanks for clarifying, @farzher. 👍
